internal/svcmanager/svcmanager.go

```diff
@@ -0,0 +1,96 @@
+package svcmanager
+
+import (
+	"time"
+
+	"github.com/ActiveState/cli/internal/appinfo"
+	"github.com/ActiveState/cli/internal/config"
+	"github.com/ActiveState/cli/internal/constants"
+	"github.com/ActiveState/cli/internal/errs"
+	"github.com/ActiveState/cli/internal/exeutils"
+	"github.com/ActiveState/cli/internal/fileutils"
+	"github.com/ActiveState/cli/internal/locale"
+	"github.com/ActiveState/cli/internal/logging"
+	"github.com/ActiveState/cli/pkg/platform/api/svc"
+	"github.com/ActiveState/cli/pkg/platform/model"
+	"golang.org/x/net/context"
+)
+
+type Manager struct {
+	ready bool
+	cfg   *config.Instance
+}
+
+func New(cfg *config.Instance) *Manager {
+	mgr := &Manager{false, cfg}
+	return mgr
+}
+
+func (m *Manager) Start() error {
+	svcInfo := appinfo.SvcApp()
+	if !fileutils.FileExists(svcInfo.Exec()) {
+		return errs.New("Could not find: %s", svcInfo.Exec())
+	}
+
+	if _, err := exeutils.ExecuteAndForget(svcInfo.Exec(), []string{"start"}); err != nil {
+		return errs.Wrap(err, "Could not start %s", svcInfo.Exec())
+	}
```

Comment:
I am realizing that we will need an IPC lock ie., a file locking mechanism for this block: https://github.com/ActiveState/cli/blob/9498ec5a8029d4b01c15514db21bcb84dad5d615/cmd/state-svc/manager.go#L31-L51

If two managers are started at the same time, they both might find out that the State Tool Service is not running yet, before they write their newly started process's PID in the config file.  Then, we end up with two state-svc services at the same time.

This will also create problem with updates, as one of the State Tool processes will be stray, as its PID has been overwritten in the configuration file.

Reply:
I'll file a story, feel like that's separate from what this PR is trying to do.

Reply:
https://www.pivotaltracker.com/story/show/178150478